### ISSUE TITLE
StaticLambdaBindingInspection & StaticClosureCanBeUsedInspection inspection context improvement

### ISSUE THREAD

**Issue by ivanpepelko** [kalessil/phpinspectionsea#1636]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) or Php Inspections (EA Ultimate), which version |
| Language level | e.g. PHP 7.4 (check project settings or composer.json)                        |

Suggestion to improve  StaticLambdaBindingInspection & StaticClosureCanBeUsedInspection inspections.
Currently they are not aware of their scope (global, function, OO, static OO), which triggers them unnecesarily.

Additionally, in OO static scope inside the closure, usage of $this is not warned about.

The following code sample illustrates the issue:
```php
<?php

// global/function scope

// inconsistent with $c3, not an object scope; this behaviour is expected in this context 
$c1 = function () { var_dump(['not static global scope' => isset($this)]); };

// triggers StaticLambdaBindingInspection (on variable `$this`), makes no sense here - $this is not bound anyway
$c2 = static function () { var_dump(['static global scope' => isset($this)]); };

// triggers StaticClosureCanBeUsedInspection (on keyword `function`), makes no sense here - closure will not be bound to an object
$c3 = function () { var_dump(['should not ask to declare closure static']); };

// same as global scope
function context()
{
    $c1 = function () { var_dump(['not static global scope' => isset($this)]); };
    $c2 = static function () { var_dump(['static global scope' => isset($this)]); };
    $c3 = function () { var_dump(['should not ask to declare closure static']); };
}


// OO scope
$class = new class () {
    // 
    public function context(): void
    {
        $c1 = function () { var_dump(['not static object scope' => isset($this)]); };

        // trigger StaticLambdaBindingInspection (on var `$this`) only in this context (current behaviour)
        $c2 = static function () { var_dump(['static object scope' => isset($this)]); };

        // trigger StaticClosureCanBeUsedInspection (on keyword `function`) only in this context (current behaviour)
        $c3 = function () { var_dump(['should ask to declare closure static']); };
    }

    public static function staticContext(): void
    {
        // ok: Using $this when not in object context
        $this->context();

        $c1 = function () {
            var_dump(
                [
                    'not static object scope' => isset($this),
                    // no warning about $this in static context
                    $this->context(),
                ]
            );
        };

        // trigger StaticLambdaBindingInspection (on var `$this`) only in this context (current behaviour)
        $c2 = static function () { var_dump(['static object scope' => isset($this)]); };

        // trigger StaticClosureCanBeUsedInspection (on keyword `function`) only in this context (current behaviour)
        $c3 = function () { var_dump(['should ask to declare closure static']); };
    }
};

```

This is something that should be improved upon in future versions.
